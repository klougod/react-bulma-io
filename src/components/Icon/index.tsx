import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export const Icon = ({ icon }: { icon: any }) => (
  <span className='icon is-small is-left'>
    <FontAwesomeIcon icon={icon} />
  </span>
)

Icon.defaultProps = {
  icon: null
}

export default Icon
