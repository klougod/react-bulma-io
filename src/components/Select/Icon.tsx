import React from 'react'
import PropTypes from 'prop-types'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

export const Icon = ({ icon }: any) => (
  <span className='icon is-small is-left'>
    <FontAwesomeIcon icon={icon} />
  </span>
)

Icon.defaultProps = {
  icon: null
}

Icon.propTypes = {
  icon: PropTypes.any
}

export default Icon
