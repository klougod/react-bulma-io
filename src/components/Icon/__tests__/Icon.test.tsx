import React from 'react'
import Enzyme from 'enzyme'
import { faAdjust } from '@fortawesome/free-solid-svg-icons'

import Icon from '../'

describe('Icon component test', function () {
  it('should render without crash with icon', function () {
    const iconMock = Enzyme.shallow(<Icon icon={faAdjust} />)
    expect(iconMock).toMatchSnapshot()
  })
})
