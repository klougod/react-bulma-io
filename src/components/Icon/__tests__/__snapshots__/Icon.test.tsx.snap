// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component test should render without crash with icon 1`] = `
<span
  className="icon is-small is-left"
>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon={
      Object {
        "icon": Array [
          512,
          512,
          Array [],
          "f042",
          "M8 256c0 136.966 111.033 248 248 248s248-111.034 248-248S392.966 8 256 8 8 119.033 8 256zm248 184V72c101.705 0 184 82.311 184 184 0 101.705-82.311 184-184 184z",
        ],
        "iconName": "adjust",
        "prefix": "fas",
      }
    }
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    swapOpacity={false}
    symbol={false}
    title=""
    transform={null}
  />
</span>
`;
